src/gui/mzroll/peakdetectiondialog.h

```diff
@@ -25,6 +25,7 @@ class PeakDetectionDialog : public QDialog, public Ui_PeakDetectionDialog
 				 ~PeakDetectionDialog();
 				 void setSettings(QSettings* settings) { this->settings = settings; }
 				 void setMainWindow(MainWindow* w) { this->mainwindow = w; }
+                 MainWindow* getMainWindow() {return mainwindow;}
```

Comment:
Is it a good idea to have Mainwindow pointer here? If no, are we planning to move?

Reply:
The child widgets generally know who their parent is, in this case it's mainwindow.  I think there should be a class called 'Toolbar'  or maybe something else , and that should ideally be the parent of all the dialogs we have. 
Having everything under mainwindow is not good. For now we don't have better solution

Reply:
We can change that later then.